tasks/assign_pr.py

```diff
@@ -0,0 +1,79 @@
+import json
+import os
+import re
+from collections import Counter
+
+from invoke import task
+
+from tasks.libs.pipeline_notifications import read_owners
+
+
+@task
+def add_team_labels(ctx, pr_id):
+    """
+    Add team labels to a PR based on the changed dependencies
+    """
+
+    codeowners = read_owners(".github/CODEOWNERS")
+
+    # Get what was changed according to PR title
+    title_words = ctx.run(f"gh pr view {pr_id} | head -n 1", hide=True).stdout.split()
+    dependency = title_words[2]
+    if "group" in title_words:  # dependabot defines group. Currently dep name is part of the group name
+        group_index = title_words.index("group")
+        dependency = title_words[group_index - 1]
+    import_module = re.compile(rf"^[ \t]*\"{dependency}.*$")
+    folder = f"./{title_words[-1][1:]}" if title_words[-2] == "in" else "."
+
+    # Find the responsible person for each file
+    owners = []
+    for root, _, files in os.walk(folder):
+        if root == "./.git" or any(root.startswith(prefix) for prefix in ["./venv", "./.git/"]):
+            continue
+        for file in files:
+            file_path = os.path.join(root, file)
+            if "doc" in file_path.casefold():
+                continue
+            with open(file_path, "r") as f:
+                try:
+                    for line in f:
+                        if is_go_module(dependency):
+                            if import_module.match(line):
+                                owners.extend([owner[1] for owner in codeowners.of(file_path[2:])])
+                                break
+                        elif dependency in line:
+                            owners.extend([owner[1] for owner in codeowners.of(file_path[2:])])
+                            break
+                except UnicodeDecodeError:
+                    continue
+    c = Counter(owners)
+    responsible = c.most_common(1)[0][0].replace('@Datadog/', '').replace('@DataDog/', '')
+    # Hardcode for USM as owner name does not match team label name
+    if responsible == "universal-service-monitoring":
+        responsible = "usm"
+
+    # Retrieve & assign labels
+    team_labels = [team["name"] for team in json.loads(ctx.run("gh label list -S team --json name", hide=True).stdout)]
+    try:
+        label = next(label for label in team_labels if responsible in label)
+    except StopIteration:
+        label = "team/agent-platform"
+    ctx.run(f"gh pr edit {pr_id} --remove-label \"team/triage\"")
+    ctx.run(f"gh pr edit {pr_id} --add-label \"{label}\"")
```

Comment:
I absolutely don't mind switching to the `gh` cli tool instead of using their API, but we might want to pick one and stick with it

Reply:
it was just easier for such actions :/

Reply:
Oh don't get me wrong I'm not saying you must use the API here because we use it elsewhere! But there are probably other places where using the `gh` cli would be simpler.
This is definitely not a blocking comment

Reply:
I'd support an effort to move to `gh`

Reply:
Real question: does `gh` provide everything the API does ?
I had some fun writing scripts to get some (relatively) complex info about PRs recently and I'm curious if I could have just used `gh` instead :smile:
FYI I used https://github.com/PyGithub/PyGithub which provides super convenient wrappers around the API, everything is typed, fetched lazily, it handles pagination, etc